### PR TITLE
Simplified MessageDetails toggle logic.

### PR DESCRIPTION
**Pull request by U-DON**
Refactored MessageDetails so that instead of using hidden inputs, it uses state and CSS classes to show and hide message details. Also added the corresponding unit test.

### CODE REVIEW

File: src/js/messaging/components/MessageDetails.jsx [hunk 1]
```diff
@@ -76,35 +76,34 @@ class MessageDetails extends React.Component {
...
     );
 
     return (
       <div
-          ref="detailsControl"
           className="messaging-message-details-control"
           tabIndex="-1"
-          onBlur={this.hideDetails}
+          onBlur={() => {
+            if (this.state.expandedFull) {
```
Comment (bshyong) [inline src/js/messaging/components/MessageDetails.jsx]: if the purpose of this is to always hide the details on blur, would it be worth doing something like `this. toggleFullDetails(false)`?, i.e. we can override the toggle action by passing a param (true or false).

**Review by bshyong — APPROVED** [review]
LGTM w/ a minor comment

File: src/js/messaging/components/MessageDetails.jsx (hunk 1 above)
Comment (U-DON) [inline src/js/messaging/components/MessageDetails.jsx]: Hm, in this case, maybe it's better to just directly set the state than calling a method named `toggle*` if it's not really toggling...

Comment (bshyong) [inline src/js/messaging/components/MessageDetails.jsx]: Yup I think I've run into this situation before: it's an action that is used as a toggle in one case, but also reused to solely open/close in another case..

Really minor issue thought, this PR LGTM as is

**Pull request merged**